### DISCUSSION TITLE
Displaying Data on a Map

### DISCUSSION THREAD

**Discussion by Talia22** [nasa/openmct#5710] [Community Q&A] [answered]
I'm using OpenMCT to present real-time flight data. What would be the current best practices exist to present this data on a 2D/3D map? Is there a recommended plugin? interface to external mapping solutions like Google Earth or Cesium etc.
Thanks in advance,
Talia

**Comment by akhenry** [comment, marked as answer]
Hi there. Right now there is no integration with mapping tools out of the box. We are working on an open source integration with [JPL's MMGIS tool](https://github.com/NASA-AMMOS/MMGIS), so look out for that. It's based on leaflet.

A few years ago we did an integration with Open Layers as well, [which is publicly available on github](https://github.com/nasa/openmct-map). Because of the MMGIS integration we don't use this any more, so due to code rot it may not just work out of the box, but I think its still a pretty good guide for integrating Open MCT with a mapping tool.

There is also an [even older experiment with Cesium](https://github.com/nasa/openmct-cesium) which bootstraps a cesium view within Open MCT but doesn't do much else.
